addons/flannel/0.21.5/install.sh

```diff
@@ -148,8 +148,38 @@ function flannel_ready_spinner() {
     fi
 }
 
+# check if weave is installed fully or partially
 function flannel_weave_conflict() {
-    ls /etc/cni/net.d/*weave* >/dev/null 2>&1
+    # Check for weave files in /etc/cni/net.d and /opt/cni/bin
+    if ls /etc/cni/net.d/*weave* >/dev/null 2>&1 || ls /opt/cni/bin/*weave* >/dev/null 2>&1; then
+        return 1
+    fi
+
```

Comment:
does anything in the upgrade process ever remove the old weave CNI binaries...? should it?

I wonder if some kind of temp file that gets created somewhere independently at the start of the migration, and gets removed right on the end of the migration is a better approach...? then if the temp file is found there on the next execution, it would detect and re-run it

just avoids the situation of the migration attempt running after it's 100% successful

Reply:
the other bonus: if something doesn't go to plan even after it thinks it was successful, you can always just recreate that temp file manually and run an install to trigger it again...

I remember when trying to get things to operate on a broken cluster yesterday trying to create a file called `/etc/cni/net.d/weave` but that wasn't enough

Reply:
I agree with your comment.
I think we should re-design this one. 

I am closing this.